### DISCUSSION TITLE
Keycloak with React integration facing refresh issue

### DISCUSSION THREAD

**Discussion by Ruptam** [keycloak/keycloak#19952] [Q&A] [answered]
Hello Everyone,

I am new to keycloak and trying to achieve PKCE based login using keycloak & react.
For that I have created a new keycloak realm and a new client with following properties:
1. Access Type: Public
2. Standard flow: on
3. Valid redirect url: http://localhost:3000/*
4. Web origin: +
5. In the advance setting chose PKCE as S256

For react application I have installed
1. keycloak-js: 21.1.0-1
2. react-keycloak/web: 3.4.0

I have added Keycloak.js and the content is:

![Screenshot 2023-04-25 235820](https://user-images.githubusercontent.com/32136351/234368727-534e4ef7-55c8-4d39-9189-8693af522a31.png)

My App.js looks like this
![Screenshot 2023-04-25 235638](https://user-images.githubusercontent.com/32136351/234368451-ed5628a1-bef4-494b-8300-ddca8afac614.png)

Now when I am accessing the react application with the url http://localhost:3000/ I am able to access the home screen but the url I can see is refreshing continously.
I have tried to attach a video how it is working.

https://user-images.githubusercontent.com/32136351/234369743-363b3a4f-dfc0-4ec9-8cfd-c664148eacab.mp4

Can anybody help me in this regards where it is going wrong.

**Comment by Paul6552** [comment, marked as answer]
First dont use the library "react-keycloak/web" it is deprecated, 3 years old and not needed

https://youtu.be/q50LxyGtEf0

**Comment by jonkoops** [comment]
This is likely a duplicate of #19452
